web-ui/src/main/resources/catalog/views/default/module.js

```diff
@@ -413,9 +413,12 @@
         addMdLayerToMap: function (link, md) {
           // This is probably only a service
           // Open the add service layer tab
-          $location.path("map").search({
-            add: encodeURIComponent(angular.toJson([buildAddToMapConfig(link, md)]))
-          });
+          var config = buildAddToMapConfig(link, md);
+          if (angular.isObject(config)) {
+            $location.path('map').search({
+              add: encodeURIComponent(angular.toJson(
+                [config]))});
+          }
           return;
         },
         addAllMdLayersToMap: function (layers, md) {
```

Comment:
`buildAddToMapConfig`  is also used in this function and the logic doesn't seem it will work fine when using an external viewer. 

Not a problem of this pull request, but the logic `buildAddToMapConfig` it's not very consistent in my opinion: returns a configuration value, when using the internal viewer, otherwise when using an external viewer executes and action, opening the viewer.

Also when adding several layers and using a external viewer, the logic in `addAllMdLayersToMap` would try to open the viewer multiple times, if it would work, that I doubt it.

Reply:
No the external viewer will be opened once and then the `_toView` array is emptied, so every subsequent call to `_commit` will have no effect.

Reply:
The `_toView` array contains all the layers that should be opened in the external viewer, that's it! every time `viewService` is called, a layer is added in `_toView` and on the first call to `_commit` the external viewer is opened with all the layers.